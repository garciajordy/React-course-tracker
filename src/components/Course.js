import React from 'react'
import { Link } from "react-router-dom";

function Course({ handleClick, course }) {
    return (
        <div className="course-component">
            <Link onClick={()=>handleClick(course)} to={"/course"}>
            { course.name }
            </Link>
        </div>
    )
}

export default Course
