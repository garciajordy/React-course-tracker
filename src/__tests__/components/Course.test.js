import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Course from '../../components/Course';
import store from '../../store/Store';

afterEach(cleanup);

function test() {
  return true;
}
const courseName = 'Ruby';
const courseId = 5;
it('should take a snapshot', () => {
  const { asFragment } = render(
    <BrowserRouter>
      <Provider store={store}>
        <Course handleClick={test} courseName={courseName} courseId={courseId} />
      </Provider>
    </BrowserRouter>,
  );
  expect(asFragment(
    <Provider store={store}>
      <Course handleClick={test} courseName={courseName} courseId={courseId} />
    </Provider>,
  )).toMatchSnapshot();
});

it('displays heading', () => {
  render(
    <BrowserRouter>
      <Provider store={store}>
        <Course handleClick={test} courseName={courseName} courseId={courseId} />
      </Provider>
    </BrowserRouter>,
  );
  expect(screen.getByText('Ruby')).toBeTruthy();
});
