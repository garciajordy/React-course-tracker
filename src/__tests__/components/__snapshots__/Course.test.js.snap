// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="course-component"
  >
    <a
      href="/course"
    >
      Ruby
    </a>
  </div>
</DocumentFragment>
`;
